LINVAST.Tests/Imperative/Builders/Go/BuildingErrorTests.cs

```diff
@@ -0,0 +1,67 @@
+    {
+        [Test]
+        public void SourceNotFoundTest()
+        {
+            Assert.Throws<FileNotFoundException>(() => new ImperativeASTFactory().BuildFromFile("404.go"));
+        }
+
+        [Test]
+        public void InvalidDeclarationTests()
+        {
+            this.AssertThrows<SyntaxErrorException>("func f int { }", p => ((GoParser)p).functionDecl());
+            this.AssertThrows<SyntaxErrorException>("func ()", p => ((GoParser)p).functionDecl());
+            this.AssertThrows<SyntaxErrorException>("func f(0, x) int { }", p => ((GoParser)p).functionDecl());
+            this.AssertThrows<SyntaxErrorException>("func f(3) int { }", p => ((GoParser)p).functionDecl());
+            this.AssertThrows<SyntaxErrorException>("func f(int x[]) int { }", p => ((GoParser)p).functionDecl());
+            this.AssertThrows<SyntaxErrorException>("func f[](x int,) int { }", p => ((GoParser)p).functionDecl());
+            this.AssertThrows<SyntaxErrorException>("func f(int x, int y,,) int { }", p => ((GoParser)p).functionDecl());
+            this.AssertThrows<SyntaxErrorException>("var x = ;;", p => ((GoParser)p).varDecl());
+            // this.AssertThrows<SyntaxErrorException>("var x int = .3, 2..", p => ((GoParser)p).varDecl());
+            this.AssertThrows<SyntaxErrorException>("var x int = ..3", p => ((GoParser)p).varDecl());
+            this.AssertThrows<SyntaxErrorException>("var x int = ()", p => ((GoParser)p).varDecl());
+            this.AssertThrows<SyntaxErrorException>("\"math\"", p => ((GoParser)p).importDecl());
+            this.AssertThrows<SyntaxErrorException>("import", p => ((GoParser)p).importDecl());
+            this.AssertThrows<SyntaxErrorException>("import * \"math\" ", p => ((GoParser)p).importDecl());
+        }
+
+        [Test]
+        public void InvalidIfStatementTests()
+        {
+            this.AssertThrows<SyntaxErrorException>("if x ", p=>((GoParser)p).ifStmt());
+            this.AssertThrows<SyntaxErrorException>("if {x} {} else {} ", p=>((GoParser)p).ifStmt());
+            this.AssertThrows<SyntaxErrorException>("if x then { } else { } ", p=>((GoParser)p).ifStmt());
+            this.AssertThrows<SyntaxErrorException>("if (x > 1 {} ", p=>((GoParser)p).ifStmt());
+            this.AssertThrows<SyntaxErrorException>("if 1 ;; else ; ", p=>((GoParser)p).ifStmt());
+        }
+
+        [Test]
+        public void InvalidForStatementTests()
+        {
+            this.AssertThrows<SyntaxErrorException>("for (x := 0; x < 5; x++) {}", p=>((GoParser)p).forStmt());
+            this.AssertThrows<NotImplementedException>("for x := 0; x < 5; x++ {}", p=>((GoParser)p).forStmt());
+            this.AssertThrows<SyntaxErrorException>("for (;;;;){}", p=>((GoParser)p).forStmt());
+        }
+
+
+        protected override ASTNode GenerateAST(string src)
+            => throw new NotImplementedException("Use GenerateAST(string, Func<Parser, ParserRuleContext) instead; " +
```

Comment:
That sounds feasible. I suppose it would be declaration node? What fields should it contain, i.e. should it only have a single string (package/module name), or a collection/array (to denote that it's usually hierarchical)?

Reply:
I would say something similar to [ImportNode](https://github.com/LINVAST/LINVAST/blob/master/LINVAST.Imperative/Nodes/ImportNodes.cs). A single string should be fine for now, no need to overcomplicate yet.

Reply:
Addressed by [3e12180](https://github.com/LINVAST/LINVAST/pull/6/commits/3e121802ef6e26b82aedcabd7886f09121f21eb5), [fd5bed4](https://github.com/LINVAST/LINVAST/pull/6/commits/fd5bed4a3dcf18cfca7e4db3e3914e35b689ddd6) (fixed failing tests in subsequent commits).